lib/encode-form-data.js

```diff
@@ -16,29 +16,31 @@ var validContentTypes = [
   'image/gif'
 ];
 
+function append (form, item, data, contentType) {
+  var options = item.options;
+  var len = typeof data === 'string' ? Buffer.byteLength(data) : data.length;
+  var header = '--' + form.getBoundary() + CRLF +
+    'Content-Type: ' + contentType + CRLF +
+    'Content-Disposition: form-data' +
+    (item.filename ? '; filename=' + encodeURIComponent(options.filename) : '') +
+    (item.name ? '; name=' + encodeURIComponent(item.name) : '') +
+    '; size=' + len + CRLF + CRLF;
+
+  form.append(item.name || item.filename, data,
+    objectAssign({ header: header, knownLength: len }, options));
+}
+
```

Comment:
@iefserge made this change since I think it was a bit hard to get the flow.

* append looked like an async method (it called `callback()`) when it wasn't
* We were overwriting input (`item.contentType` was overwritten)
* We overwrote a scoped variable (`value`) that I easily missed.

So that's the reason I made this change :)